src/Abp.Web.Mvc/Web/Mvc/Threading/HttpContextCancellationTokenProvider.cs

```diff
@@ -2,11 +2,27 @@
 using Abp.Threading;
 using System.Threading;
 using System.Web;
+using Abp.Runtime;
 
 namespace Abp.Web.Mvc.Threading
 {
-    public class HttpContextCancellationTokenProvider : ICancellationTokenProvider, ITransientDependency
+    public class HttpContextCancellationTokenProvider : CancellationTokenProviderBase, ITransientDependency
     {
-        public CancellationToken Token => HttpContext.Current?.Response.ClientDisconnectedToken ?? CancellationToken.None;
+        public override CancellationToken Token
+        {
+            get
+            {
+                if (OverridedValue != null)
+                {
+                    return OverridedValue.CancellationToken;
+                }
+                return HttpContext.Current?.Response.ClientDisconnectedToken ?? CancellationToken.None;
```

Comment:
@ismcagdas this probably will have the same problem with https://github.com/aspnetboilerplate/aspnetboilerplate/pull/5381 if anyone is reading from repository during application start up